rotkehlchen/exchanges/bit2me.py

```diff
@@ -0,0 +1,927 @@
+        brokerage_trades = self._query_brokerage_trades(start_ts, end_ts)
+        events.extend(brokerage_trades)
+
+        # Get exchange trades from /v1/trading/trade
+        trades, _ = self.query_online_trade_history(start_ts, end_ts)
+        events.extend(trades)
+
+        return events, end_ts
+
+    @protect_with_lock()
+    @cache_response_timewise()
+    def query_balances(self) -> ExchangeQueryBalances:
+        """Query Bit2me balances for all assets."""
+        try:
+            response = self._api_query('balances')
+        except RemoteError as e:
+            msg = f'Bit2me API request failed. {e!s}'
+            log.error(msg)
+            return None, msg
+
+        try:
+            response_data = jsonloads_list(response.text)
+        except JSONDecodeError as e:
+            msg = f'Bit2me returned invalid JSON response. {e!s}'
+            log.error(msg)
+            return None, msg
+
+        # Bit2me API returns a list of balances
+        # Each item has: currency, balance, blockedBalance, blockedOutputBalance
+        # response_data is already validated as a list by jsonloads_list
+
+        assets_balance: dict[AssetWithOracles, Balance] = {}
+
+        for balance_entry in response_data:
+            try:
+                # Get asset symbol (API field is 'currency')
+                asset_symbol = balance_entry.get('currency')
+                if not asset_symbol:
+                    log.warning(f'Bit2me balance entry missing currency symbol: {balance_entry}')
+                    continue
+
+                # Calculate total balance (available + blocked amounts)
+                balance_amount = deserialize_fval(
+                    balance_entry.get('balance', '0'),
+                    'balance',
+                    'bit2me',
+                )
+                blocked_balance = deserialize_fval(
+                    balance_entry.get('blockedBalance', '0'),
+                    'blockedBalance',
+                    'bit2me',
+                )
+                blocked_output_balance = deserialize_fval(
+                    balance_entry.get('blockedOutputBalance', '0'),
+                    'blockedOutputBalance',
+                    'bit2me',
+                )
+
+                total_amount = balance_amount + blocked_balance + blocked_output_balance
+
+                # Skip zero balances
+                if total_amount == ZERO:
+                    continue
+
+                # Convert asset symbol to rotki asset
+                try:
+                    asset = asset_from_bit2me(asset_symbol)
+                except (UnknownAsset, UnsupportedAsset) as e:
+                    self.msg_aggregator.add_warning(
+                        f'Found unsupported Bit2me asset {asset_symbol}. '
+                        f'{e!s}. Ignoring its balance.',
+                    )
+                    continue
+
+                assets_balance[asset] = Balance(amount=total_amount)
```

Comment:
You need to also find the price in the main currency. Check how we do it here https://github.com/rotki/rotki/blob/d9058c5afcb72dbea2cf0c78e0578ea51db1a5ed/rotkehlchen/exchanges/bybit.py#L450